distributed/tests/test_executor.py

```diff
@@ -3213,9 +3213,12 @@ def test_publish_simple(s, a, b):
     yield f._start()
 
     data = yield e._scatter(range(3))
-    yield e._publish_dataset(data, 'data')
-    assert 'data' in s.published_data
-    assert e.id in s.published_data['data']['clients']
+    out = yield e._publish_dataset(data, 'data')
+    assert out['success'] == 'true'
```

Comment:
I suggest that, on the client side this return None if successful and raise a `KeyError(name)` otherwise.